### PR TITLE
Add a search bar to link the original Pages to the searchable Pages

### PR DESCRIPTION
**Pull request by brogan20**
## What does this PR do?
Improve repo

## Description
Adds links to the new free-programming-books-search site and a search bar to make it possible to search from the original website.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -182,8 +201,6 @@ Free Podcasts and Screencasts:
 
 ### Programming Playgrounds
 
-Write, compile and run your code within a browser. Try it out!
```
Comment (eshellman) [inline README.md:185]: did you mean to remove this?

File: README.md [hunk 2]
```diff
@@ -8,15 +8,28 @@
...
+This page is available as an easy-to-read website at [![https://ebookfoundation.github.io/free-programming-books/](https://img.shields.io/website?style=flat&logo=www&logoColor=whitesmoke&label=https%3A%2F%2Febookfoundation.github.io&down_color=red&down_message=down&up_color=green&up_message=up&url=https%3A%2F%2Febookfoundation.github.io%2Ffree-programming-books%2F)](https://ebookfoundation.github.io/free-programming-books/). Also try our new dynamic version of the website at [![https://ebookfoundation.github.io/free-programming-books-search/](https://img.shields.io/website?style=flat&logo=www&logoColor=whitesmoke&label=https%3A%2F%2Febookfoundation.github.io/free-programming-books-search&down_color=red&down_message=down&up_color=green&up_message=up&url=https%3A%2F%2Febookfoundation.github.io%2Ffree-programming-books-search%2F)](https://ebookfoundation.github.io/free-programming-books-search/).
```
Comment (eshellman) [inline README.md:11]: Rendering looks cramped .  Maybe start a new paragraph? (not specific to this PR, but I wonder whether it's obvious to people that the button is a link? Try to word the text so it won't grow stale in a month or two; I think we want to eventually remove or de-emphasize the static site. Also not specific to this PR, but with two versions of the site, maybe we can change the anchor text to "static site" and "Dynamic site"? in stead of an impenetrable url?

File: README.md [hunk 3]
```diff
@@ -8,15 +8,28 @@
...
+This page is available as an easy-to-read website at [![https://ebookfoundation.github.io/free-programming-books/](https://img.shields.io/website?style=flat&logo=www&logoColor=whitesmoke&label=https%3A%2F%2Febookfoundation.github.io&down_color=red&down_message=down&up_color=green&up_message=up&url=https%3A%2F%2Febookfoundation.github.io%2Ffree-programming-books%2F)](https://ebookfoundation.github.io/free-programming-books/). Also try our new dynamic version of the website at [![https://ebookfoundation.github.io/free-programming-books-search/](https://img.shields.io/website?style=flat&logo=www&logoColor=whitesmoke&label=https%3A%2F%2Febookfoundation.github.io/free-programming-books-search&down_color=red&down_message=down&up_color=green&up_message=up&url=https%3A%2F%2Febookfoundation.github.io%2Ffree-programming-books-search%2F)](https://ebookfoundation.github.io/free-programming-books-search/). 
 
+<div align="center">
```
Comment (eshellman) [inline README.md:13]: does this render?

File: README.md [hunk 4]
```diff
@@ -8,15 +8,28 @@
...
+  <form action="https://ebookfoundation.github.io/free-programming-books-search">
+    <!-- <label for="search">Search: </label> -->
+    <input type="text" id="fpbSearch" name="search" placeholder="Search Book or Author"/>
+    <label for="submit"> </label>
+    <input type="submit" name="submit" value="Search" />
+  </form>
+<div>
+
+
+<div align="left" markdown="1">
```
Comment (eshellman) [inline README.md:23]: please explain the function of the extra markup.

**Review by eshellman — CHANGES_REQUESTED** [review]
Exciting!!!

File: README.md (hunk 1 above)
Comment (davorpa) [reply README.md:185]: I think so too. Restore it according to #6819

File: README.md (hunk 2 above)
Comment (davorpa) [reply README.md:11]: Keeping it simple like Eric says...

```suggestion
This page is available as an easy-to-read websites. Access them clicking on [![https://ebookfoundation.github.io/free-programming-books/](https://img.shields.io/website?style=flat&logo=www&logoColor=whitesmoke&label=Static%20site&down_color=red&down_message=down&up_color=green&up_message=up&url=https%3A%2F%2Febookfoundation.github.io%2Ffree-programming-books%2F)](https://ebookfoundation.github.io/free-programming-books/) or [![https://ebookfoundation.github.io/free-programming-books-search/](https://img.shields.io/website?style=flat&logo=www&logoColor=whitesmoke&label=Dynamic%20search%20site&down_color=red&down_message=down&up_color=green&up_message=up&url=https%3A%2F%2Febookfoundation.github.io%2Ffree-programming-books-search%2F)](https://ebookfoundation.github.io/free-programming-books-search/) links. 
```

File: README.md (hunk 3 above)
Comment (davorpa) [reply README.md:13]: Due to security reasons, forms usually don't renders on github markdown

File: README.md (hunk 4 above)
Comment (davorpa) [reply README.md:23]: I also think that doesn't is needed because `left` is the default behaviour

File: README.md (hunk 2 above)
Comment (LuigiImVector) [reply README.md:11]: By writing the links users might not understand, wouldn't it be better to replace them with simple words?

File: README.md (hunk 1 above)
Comment (brogan20) [reply README.md:185]: This was on accident. Just fixed it.

File: README.md (hunk 4 above)
Comment (brogan20) [reply README.md:23]: The extra div tags are to ensure the rendered site displays correctly. Without the align attribute the block would become centered. The markdown attribute is also needed or the block is displayed as plaintext markdown.

**Comment by CleanMachine1** [comment]
Personally not a fan of the badges being used as links within a paragraph, seems out of place.

File: README.md (hunk 2 above)
Comment (CleanMachine1) [reply README.md:11]:
```suggestion
This page is available as an [easy-to-read website](https://ebookfoundation.github.io/free-programming-books/). Also try our new [dynamic version of the website](https://ebookfoundation.github.io/free-programming-books-search/). 
```

Comment (CleanMachine1) [reply README.md:11]: Also, `dynamic version of the website` is a little vague, I'd include the fact that it has a working search tool.

Comment (LuigiImVector) [reply README.md:11]: what if we used the official 'free-programming-books-search' github pages instead?

File: README.md [hunk 5]
```diff
@@ -8,15 +8,28 @@
...
 
+<div align="center">
+  <form action="https://ebookfoundation.github.io/free-programming-books-search">
+    <!-- <label for="search">Search: </label> -->
+    <input type="text" id="fpbSearch" name="search" placeholder="Search Book or Author"/>
+    <label for="submit"> </label>
+    <input type="submit" name="submit" value="Search" />
+  </form>
```
Comment (davorpa) [inline README.md:19]: attribute `for` always goes against target element `id`. Another way to solve without ids is wrapping the `input` into the `label`

```suggestion

  <form action="https://ebookfoundation.github.io/free-programming-books-search">
    <!-- <label for="fpbSearch">Search: </label> -->
    <input type="text" id="fpbSearch" name="search" placeholder="Search Book or Author"/>
    <label for="submit"> </label>
    <input type="submit" id="submit" name="submit" value="Search" />
  </form>

```

**Review by davorpa — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (davorpa) [reply README.md:11]: > what if instead we use 'free-programming-books-search' as the (only) official github page?

I prefer both meanwhile parser works well at 99%. At this time search doesn't detect some resources. E.g.: spanish midudev

File: README.md (hunk 5 above)
Comment (brogan20) [reply README.md]: Added the id and removed the commented out label

File: README.md (hunk 2 above)
Comment (eshellman) [reply README.md:11]: I like @davorpa 's version

Comment (brogan20) [reply README.md]: I'll commit the version suggested by @davorpa

File: README.md [hunk 6]
```diff
@@ -8,15 +8,27 @@
...
+This page is available as an easy-to-read websites. Access them clicking on [![https://ebookfoundation.github.io/free-programming-books/](https://img.shields.io/website?style=flat&logo=www&logoColor=whitesmoke&label=Static%20site&down_color=red&down_message=down&up_color=green&up_message=up&url=https%3A%2F%2Febookfoundation.github.io%2Ffree-programming-books%2F)](https://ebookfoundation.github.io/free-programming-books/) or [![https://ebookfoundation.github.io/free-programming-books-search/](https://img.shields.io/website?style=flat&logo=www&logoColor=whitesmoke&label=Dynamic%20search%20site&down_color=red&down_message=down&up_color=green&up_message=up&url=https%3A%2F%2Febookfoundation.github.io%2Ffree-programming-books-search%2F)](https://ebookfoundation.github.io/free-programming-books-search/) links.
```
Comment (davorpa) [inline README.md:11]: Remove trailing spaces

```suggestion
This page is available as an easy-to-read websites. Access them clicking on [![https://ebookfoundation.github.io/free-programming-books/](https://img.shields.io/website?style=flat&logo=www&logoColor=whitesmoke&label=Static%20site&down_color=red&down_message=down&up_color=green&up_message=up&url=https%3A%2F%2Febookfoundation.github.io%2Ffree-programming-books%2F)](https://ebookfoundation.github.io/free-programming-books/) or [![https://ebookfoundation.github.io/free-programming-books-search/](https://img.shields.io/website?style=flat&logo=www&logoColor=whitesmoke&label=Dynamic%20search%20site&down_color=red&down_message=down&up_color=green&up_message=up&url=https%3A%2F%2Febookfoundation.github.io%2Ffree-programming-books-search%2F)](https://ebookfoundation.github.io/free-programming-books-search/) links.
```

**Review by davorpa — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (davorpa) [reply README.md:22]: The problem is on line 20. `<div>` tag enclosing `form` is poor closed 😉. Then neither of other `<div align="left" markdown="1">` are needed

File: README.md [hunk 7]
```diff
@@ -8,15 +8,27 @@
...
 
+<div align="center">
+  <form action="https://ebookfoundation.github.io/free-programming-books-search">
+    <input type="text" id="fpbSearch" name="search" placeholder="Search Book or Author"/>
+    <label for="submit"> </label>
+    <input type="submit" id="submit" name="submit" value="Search" />
+  </form>
+<div>
```
Comment (davorpa) [inline README.md:19]: Block layer is closed wrong

```suggestion
</div>
```

Then neither of the extra `<div align="left" markdown="1">` are needed :wink:

**Review by davorpa — CHANGES_REQUESTED** [review]

File: README.md [hunk 8]
```diff
@@ -182,8 +200,8 @@ Free Podcasts and Screencasts:
 
 ### Programming Playgrounds
 
-Write, compile and run your code within a browser. Try it out!
-
+Write, compile and run your code within a browser. Try it out! 
+
```
Comment (davorpa) [inline README.md:204]: Remove unnecesary trailing whitespaces

```suggestion
Write, compile and run your code within a browser. Try it out!

```

**Review by davorpa — APPROVED** [review]
LGTM now :rocket:

Tested with https://github.com/davorpa/free-programming-books/commit/8fc239000d131e44a9b262c2d24f9f19900367de commit on my forked mirror website https://davorpa.github.io/free-programming-books/

image

**Pull request merged**